src/Test/UnitTests/basic_constraint_tests.jl

```diff
@@ -0,0 +1,233 @@
+    (MOI.ScalarQuadraticFunction{Float64}, MOI.GreaterThan{Float64}) => ( dummy_scalar_quadratic, 1, MOI.GreaterThan(1.0) ),
+    (MOI.ScalarQuadraticFunction{Float64}, MOI.EqualTo{Float64})     => ( dummy_scalar_quadratic, 1, MOI.EqualTo(1.0) ),
+    (MOI.ScalarQuadraticFunction{Float64}, MOI.Interval{Float64})    => ( dummy_scalar_quadratic, 1, MOI.Interval(1.0, 2.0) ),
+
+    (MOI.VectorAffineFunction{Float64}, MOI.Reals)        => ( dummy_vector_affine, 2, MOI.Reals(2) ),
+    (MOI.VectorAffineFunction{Float64}, MOI.Zeros)        => ( dummy_vector_affine, 2, MOI.Zeros(2) ),
+    (MOI.VectorAffineFunction{Float64}, MOI.Nonpositives) => ( dummy_vector_affine, 2, MOI.Nonpositives(2) ),
+    (MOI.VectorAffineFunction{Float64}, MOI.Nonnegatives) => ( dummy_vector_affine, 2, MOI.Nonnegatives(2) ),
+
+    (MOI.VectorQuadraticFunction{Float64}, MOI.Reals)        => ( dummy_vector_quadratic, 2, MOI.Reals(2) ),
+    (MOI.VectorQuadraticFunction{Float64}, MOI.Zeros)        => ( dummy_vector_quadratic, 2, MOI.Zeros(2) ),
+    (MOI.VectorQuadraticFunction{Float64}, MOI.Nonpositives) => ( dummy_vector_quadratic, 2, MOI.Nonpositives(2) ),
+    (MOI.VectorQuadraticFunction{Float64}, MOI.Nonnegatives) => ( dummy_vector_quadratic, 2, MOI.Nonnegatives(2) )
+)
+"""
+    basic_constraint_tests(model::MOI.ModelLike, config::TestConfig;
+        delete                  = true,
+        get_constraint_function = true,
+        get_constraint_set      = true,
+        include                 = Any[],
+        exclude                 = Any[]
+        )
+
+Test basic constraint functionality for all function-in-set combinations that
+are supported by `model`.
+
+See also `test_basic_constraint_functionality`.
+
+If `delete = true`, it will test the deletion of constraints.
+If `get_constraint_function = true`, it will test the getting of `ConstraintFunction`.
+If `get_constraint_set = true`, it will test the getting of `ConstraintSet`.
+
+`include` and `exclude` can be used to run a subset of the tests, although only
+one can be used in each call.
+
+### Examples
+
+    basic_constraint_tests(model, config; exclude = [
+        (MOI.ScalarAffineFunction{Float64}, MOI.Interval{Float64})
+    ])
+
+    basic_constraint_tests(model, config;
+        delete = false,
+        include = [
+            (MOI.VectorOfVariables, MOI.SOS2{Float64})
+        ]
+    )
+"""
+function basic_constraint_tests(model::MOI.ModelLike, config::TestConfig;
+    delete                  = true,
+    get_constraint_function = true,
+    get_constraint_set      = true,
+    include                 = Any[],
+    exclude                 = Any[]
+    )
+    if length(include) > 0 && length(exclude) > 0
+        error("Don't use `include` and `exclude` together.")
+    end
+    test_keys = length(include) > 0 ? include : Iterators.filter(x->!(x in exclude), keys(BasicConstraintTests))
```

Comment:
Why don't you do `test_keys = keys(BasicConstraintTests)` in the case where `include` and `exclude` are both empty. Since you do `if MOI.supportsconstraint` later, this would test all supported constraints and remove the need for a `ListOfSupportedConstraintTypes` discussed in https://github.com/JuliaOpt/MathOptInterface.jl/pull/357#issuecomment-388686469

Reply:
> This does test keys(BasicConstraintTests) if include and exclude are both empty.

Indeed ^^

> If we had ListOfSupportedConstraintTypes, we could throw a warning that we didn't test SingleVariable-in-LessThan{Int} for example.

The warning would be called when `include` is non-empty and it does not contain all supported sets ? This can already be done now by iterating through the keys of `BasicConstraintTests`